Unable do extract method with try and subscriptable

**Describe the bug**
Error while extracting method that contains augmented assignment to subscriptable (dict, list etc.) inside try..except.

**To Reproduce**
Steps to reproduce the behavior:
1. Code before refactoring:

```python
any_subscriptable = {0: 0}
try:
    any_subscriptable[0] += 1
except:
    pass
```

2. Describe the refactoring you want to do

Extracting method.

3. Expected code after refactoring:

```python
def extracted():
    any_subscriptable = {0: 0}
    try:
        any_subscriptable[0] += 1
    except:
        pass
extracted()
```

4. Describe the error or unexpected result that you are getting

```
self = <rope.refactor.extract._FunctionInformationCollector object at ...>, node = <ast.AugAssign object at ...>

    def _AugAssign(self, node):
        ast.walk(node.value, self)
>       self._read_variable(node.target.id, node.target.lineno)
E       AttributeError: 'Subscript' object has no attribute 'id'
```

**Editor information (please complete the following information):**
 - Project Python version: 3.10.1
 - Rope Python version: 3.10.1
 - Rope version: 0.22.0, 0.23.0

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,9 @@
 # **Upcoming release**
 
+## Bug fixes
+
+- #459 Fix bug while extracting method with augmented assignment to subscript in try block (@dryobates)
+
 # Release 0.23.0
 
 ## Syntax support

--- a/rope/refactor/extract.py
+++ b/rope/refactor/extract.py
@@ -843,8 +843,12 @@ class _FunctionInformationCollector(object):
 
     def _AugAssign(self, node):
         ast.walk(node.value, self)
-        self._read_variable(node.target.id, node.target.lineno)
-        self._written_variable(node.target.id, node.target.lineno)
+        if isinstance(node.target, ast.Subscript):
+            target_id = node.target.value.id
+        else:
+            target_id = node.target.id
+        self._read_variable(target_id, node.target.lineno)
+        self._written_variable(target_id, node.target.lineno)
 
     def _ClassDef(self, node):
         self._written_variable(node.name, node.lineno)

--- a/ropetest/refactor/extracttest.py
+++ b/ropetest/refactor/extracttest.py
@@ -1762,6 +1762,30 @@ class ExtractMethodTest(unittest.TestCase):
         """)
         self.assertEqual(expected, refactored)
 
+    def test_extract_method_and_augmentedj_assignment_in_try_block(self):
+        code = dedent("""\
+            def f():
+                any_subscriptable = [0]
+                try:
+                    any_subscriptable[0] += 1
+                except Exception:
+                    pass
+        """)
+        start, end = self._convert_line_range_to_offset(code, 2, 6)
+        refactored = self.do_extract_method(code, start, end, "g")
+        expected = dedent("""\
+            def f():
+                g()
+
+            def g():
+                any_subscriptable = [0]
+                try:
+                    any_subscriptable[0] += 1
+                except Exception:
+                    pass
+        """)
+        self.assertEqual(expected, refactored)
+
     def test_extract_and_not_passing_global_functions(self):
         code = dedent("""\
             def next(p):

